dokka-integration-tests/gradle/build.gradle.kts

```diff
@@ -177,3 +195,42 @@ val checkoutKotlinxSerialization by tasks.registering(GitCheckoutTask::class) {
     commitId = "ed1b05707ec27f8864c8b42235b299bdb5e0015c"
     destination = templateProjectsDir.dir("serialization/kotlinx-serialization")
 }
+
+testing {
+    suites.withType<JvmTestSuite>().configureEach {
+        dependencies {
+            implementation(testFixtures(project()))
+        }
+        targets.configureEach {
+            testTask.configure {
+                devMavenPublish.configureTask(this)
+            }
+        }
+    }
+    val testExampleProjects by suites.registering(JvmTestSuite::class) {
+        targets.configureEach {
+            testTask.configure {
+
+                val exampleGradleProjectsDir = projectDir.resolve("../../examples/gradle-v2")
+                systemProperty
+                    .inputDirectory("exampleGradleProjectsDir", exampleGradleProjectsDir)
+                    .withPathSensitivity(RELATIVE)
+
+                val expectedDataDir = layout.projectDirectory.dir("src/testExampleProjects/expectedData")
```

Comment:
I found the workflow of updating the files fairly easy to follow, although I'm sure it could be improved further.

Failures log a clickable link to the actual generated docs, so you just need to open the directory and drag the file into the expectedData dir.

<img width="1709" alt="image" src="https://github.com/user-attachments/assets/94fabab7-0bc8-4a19-ae7a-7b3ee4a3fe65">

Reply:
It would be nice if you could just remove the test data and rerun the test to regenerate it

Reply:
may be more safe alternative: add an explicit Gradle property to force regeneration of the output

Additionally this will be very useful, as it will be possible to force regeneration and see the diff in IDEA, and not in Gradle output (which is not very nice).
Without this, I think that the output will be not very useful, as even simple change in Dokka generation could slightly change output

Reply:
In my experience the most common DevX for test data generation is the following: when there is no test data, you run the test which writes the test data and immediately fails (with an explicit "test data generation run" message)

Reply:
Introducing machinery to automatically updating missing test data introduces a risk of the tests being silently updated and resulting in the tests passing, even if the data is outdated.

If updating the test data becomes a burden then we can look into it in the future. For now, it's not necessary.